write_test.go

```diff
@@ -225,133 +171,88 @@ func TestQueueOneParameterized(t *testing.T) {
+			t.Errorf("failed during table creation: %v - %v", err.Error(), results[0].Err.Error())
+		}
+	})
 
-	t.Logf("trying Write DROP")
-	s = make([]string, 0)
-	s = append(s, "DROP TABLE IF EXISTS "+testTableName()+"")
-	results, err = conn.Write(s)
-	if err != nil {
-		t.Logf("--> FAILED (%s)", err.Error())
-		t.Fail()
-	}
+	t.Run("Write INSERT", func(t *testing.T) {
+		s := make([]string, 0)
+		s = append(s, "INSERT INTO "+testTableName()+" (id, name) VALUES ( 21, 'aaa bbb ccc' )")
+		s = append(s, "INSERT INTO "+testTableName()+" (id, name) VALUES ( 22, 'ddd eee fff' )")
+		s = append(s, "INSERT INTO "+testTableName()+" (id, name) VALUES ( 23, 'ggg hhh iii' )")
+		s = append(s, "INSERT INTO "+testTableName()+" (id, name) VALUES ( 24, 'jjj kkk lll' )")
+		results, err := globalConnection.Write(s)
+		if err != nil {
+			for _, result := range results {
+				if result.Err != nil {
+					t.Errorf("result error: %v", result.Err)
+				}
+			}
+		}
+
+		if len(results) != 4 {
+			t.Errorf("expected results to be length of 4, got: %d", len(results))
+		}
+	})
 }
 
 func TestWriteParameterized(t *testing.T) {
-	var results []WriteResult
-	var err error
-
-	t.Logf("trying Open")
-	conn, err := Open(testUrl())
-	if err != nil {
-		t.Logf("--> FATAL")
-		t.Fatal(err)
-	}
-
-	t.Logf("trying WriteParameterized DROP & CREATE")
-	results, err = conn.WriteParameterized(
-		[]ParameterizedStatement{
-			{
-				Query: fmt.Sprintf("DROP TABLE IF EXISTS %s", testTableName()),
-			},
-			{
-				Query: fmt.Sprintf("CREATE TABLE %s (id integer, name text)", testTableName()),
-			},
-		},
-	)
-	if err != nil {
-		t.Logf("--> FAILED (%s)", err.Error())
-		t.Fail()
-	}
-
-	t.Logf("trying WriteParameterized INSERT")
-	results, err = conn.WriteParameterized(
-		[]ParameterizedStatement{
-			{
-				Query:     fmt.Sprintf("INSERT INTO %s (id, name) VALUES (?, ?)", testTableName()),
-				Arguments: []interface{}{1, "aaa bbb ccc"},
-			},
-			{
-				Query:     fmt.Sprintf("INSERT INTO %s (id, name) VALUES (?, ?)", testTableName()),
-				Arguments: []interface{}{1, "aaa bbb ccc"},
+	t.Run("DROP", func(t *testing.T) {
+		tableName := RandStringBytes(8)
+		_, err := globalConnection.WriteParameterized(
+			[]gorqlite.ParameterizedStatement{
+				{
+					Query: fmt.Sprintf("DROP TABLE IF EXISTS %s", tableName),
+				},
+				{
+					Query: fmt.Sprintf("CREATE TABLE %s (id integer, name text)", tableName),
+				},
 			},
-			{
-				Query:     fmt.Sprintf("INSERT INTO %s (id, name) VALUES (?, ?)", testTableName()),
-				Arguments: []interface{}{1, "aaa bbb ccc"},
-			},
-			{
-				Query:     fmt.Sprintf("INSERT INTO %s (id, name) VALUES (?, ?)", testTableName()),
-				Arguments: []interface{}{1, "aaa bbb ccc"},
-			},
-		},
-	)
-	if err != nil {
-		t.Logf("--> FAILED (%s)", err.Error())
-		t.Fail()
-	}
-	if len(results) != 4 {
-		t.Logf("--> FAILED, expected 4 results, got %d", len(results))
-		t.Fail()
-	}
+		)
+		if err != nil {
+			t.Logf("--> FAILED (%s)", err.Error())
```

Comment:
Let's use `t.Errorf` :)

Reply:
I thought I was gonna leave this one out for you, haha! Okay, I'll tidy this one out 😄

Reply:
Haha. Thanks!